### PR TITLE
feat: Anvil: Interact with a focused widget's widget name component

### LINKED ISSUE #33385: [Bug]: Cannot interact with widget name component when widget is focused
Fix issue where we can't click on widget name component when hovering over a widget that is not selected.

- [ ] To add a bit of consistency, please, add "grab" pointer to the title part of the name component to highlight that you can drag the handle ([original thread](https://miro.com/app/board/uXjVKWcOJsY=/?moveToWidget=3458764587329204452&cot=14))
_caveat: "clickability" of label will be lost if it's all grab_

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
- Add a ghost component (`AnvilWidgetNameComponentWrapper`) that prevents the widget underneath the widget name component from being focused
- Select widget when dragging a focused widget from the widget name component
- Adjust offsets and sizes of the ghost component and the widget name button
- Remove `onMouseLeave` events from widgets and add an `onMouseLeave` event to the Canvas.
- Change the pointer to `grab` for the widget name component

Fixes #33385

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9201483569>
> Commit: 4373df84f255534a6eb839b1cad532ae327947ec
> Cypress dashboard: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9201483569&attempt=2&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank"> Click here!</a>
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/BugTests/GitBugs_Spec.ts
> <li>cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImport_spec.ts </ol>
> To know the list of identified flaky tests - <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">Refer here</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by marks0351** [comment]
@riodeuno could you run tag.All coz of the changes to the canvas component. I am not able to assess its impact.

**Review by KelvinOm — DISMISSED** [review]

File: app/client/src/pages/Editor/Canvas.tsx [hunk 1]
```diff
@@ -93,6 +100,7 @@ const Canvas = (props: CanvasProps) => {
           )}`}
           data-testid={"t--canvas-artboard"}
           id={CANVAS_ART_BOARD}
+          onMouseLeave={unfocusAllWidgets}
```
Comment (marks0351) [inline app/client/src/pages/Editor/Canvas.tsx:103]: can we not add this on the AnvilEditorCanvas?

**Review by marks0351 — COMMENTED** [review]
have added just one comment.

File: app/client/src/pages/Editor/Canvas.tsx (hunk 1 above)
Comment (riodeuno) [reply app/client/src/pages/Editor/Canvas.tsx:103]: Sounds good. I'll add it there.

**Review by marks0351 — APPROVED** [review]

**Comment by riodeuno** [comment]
@marks0351 Tested Fixed mode hover interaction. Works fine.

**Pull request merged**